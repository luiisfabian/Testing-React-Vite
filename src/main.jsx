import React from 'react'

import ReactDom from "react-dom/client";
import { HelloWordApp } from './HelloWord';
import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';
import { CounterAppDos } from './CounterApp2';
// import {App} from '.
// import './style.css'


ReactDom.createRoot(document.getElementById('root')).render(
  
    <React.StrictMode>
        {/* <HelloWordApp title={"Hola soy Fabian"}/> */}
        {/* <FirstApp /> */}
        <CounterApp value={100}/>
        {/* <CounterAppDos value={0} /> */}
    </React.StrictMode>
)
