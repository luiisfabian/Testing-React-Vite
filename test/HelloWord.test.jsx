import { render } from "@testing-library/react";
import { HelloWordApp } from "../src/HelloWord";

describe('pruebas en HelloWord.jsx', () => {
   //  test('debe de hacer match con el snapchot', () => {
        
   //      const title = "Hola soy Fabian"
   //      const { container } = render(< HelloWordApp title={title}/> )


   //      expect(container).toMatchSnapshot();
   //   });

     test('debe mostrar el titulo en un H1', () => {
        
        const title = "Hola soy Fabian"
        const { container, getByText, getByTestId } = render(< HelloWordApp title={title}/> )
        expect(getByText(title)).toBeTruthy()

      //   const h1= container.querySelector('h1')

      //   expect(h1.innerHTML).toBe(title)

      // expect(getByTestId('test-title')).toBeTruthy();
      // toContain sirve para queno imorta lo que tenga debe contener el subtitulo
      expect(getByTestId('test-title').innerHTML).toContain(title);


     })

     test('debe mostrar el subtitulo enviado por props', () => {
        
      const title = "Hola soy Fabian"
      const subTitle = "Hola soy subtitulo"
      const { getAllByText } = render(< HelloWordApp title={title}  subTitle={subTitle}/> )

      // valido que el subtitulo exista
      // expect(getByText(subTitle)).toBeTruthy()
      expect(getAllByText(subTitle).length).toBe(2);



   })

     
});