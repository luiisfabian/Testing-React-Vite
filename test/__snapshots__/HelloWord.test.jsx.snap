// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en HelloWord.jsx debe de hacer match con el snapchot 1`] = `
<div>
  <h1>
    Hola soy Fabian
  </h1>
  <p>
    no hay subtitulo
  </p>
  <p>
    Fabian Mu√±oz
  </p>
</div>
`;
